app/api/session/session.tsx

```diff
@@ -1,6 +1,32 @@
+import "server-only";
+import { JWTPayload, SignJWT, jwtVerify } from "jose";
 import { NextResponse } from "next/server";
-export async function createSession({response, exp, session_id}:{response: NextResponse, exp: number, session_id: string}) {
-    response.cookies.set("session_id", session_id, {
+
+const sessionSecretKey = process.env.SESSION_SECRET;
+const encodedKey = new TextEncoder().encode(sessionSecretKey);
+
+export async function encrypt({ envelope, exp }:{ envelope: JWTPayload, exp: number }) {
+    return new SignJWT(envelope)
+        .setProtectedHeader({ alg: "HS256" })
+        .setIssuedAt()
+        .setExpirationTime(exp)
+        .sign(encodedKey)
+}
+ 
+export async function decrypt(session: string | undefined = "") {
+    try {
+        const { payload } = await jwtVerify(session, encodedKey, {
+            algorithms: ["HS256"],
+        })
+        return payload
+    } catch (error) {
+        console.log("Failed to verify session")
+    }
+}
```

Comment:
I am also realizing that while I am sure we want this function... we don't have any current use of it. Might make sense to put in a follow up where we at least test it.

Reply:
Removed